### PR TITLE
Initial commit of tmdb plugin.

### PR DESCRIPTION
**Pull request by cacoyle**

### CODE REVIEW

File: config-dist.json [hunk 1]
```diff
@@ -78,4 +79,8 @@
         "RESPONSE_THRESHOLD_SECS": 30,
         "COOLDOWN_SECS": 300
     }
```
Comment (cacoyle) [inline config-dist.json:81]: oic it's angry cause of a missing comma, will fix in the am

File: config-dist.json [hunk 2]
```diff
@@ -77,5 +78,9 @@
         "MAX_THRESHOLD_SECS": 40,
         "RESPONSE_THRESHOLD_SECS": 30,
         "COOLDOWN_SECS": 300
+    },
+    "TMDBPlugin": {
+      "TMDB_API_URL": "https://....",
+      "TMDB_API_KEY": "hunter2",
```
Comment (jonheese-wti) [inline config-dist.json:84]: JSON doesn't like this line having a `,` at the end when there are no entries following it.

File: plugins/tmdb.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+from collections import OrderedDict
+from urllib.parse import quote_plus
+from . import plugin
+
+import requests
+
+
+class TMDBPlugin(plugin.Plugin):
+    def __init__(self):
+        super().__init__()
```
Comment (jonheese-wti) [inline plugins/tmdb.py:10]: The way the `Plugin` class's init works, this needs to be:
```
     def __init__(self, web_client, plugin_config):
         super().__init__(web_client=web_client, plugin_config=plugin_config)
```

File: plugins/tmdb.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    def receive(self, request):
+        if super().receive(request) is False:
+            return False
+
+        if request['text'].lower().startswith("moonbeam movies"):
+            self._log.debug(f"Got movie request: {request['text']}")
+            command = request['text'].split()
+
+            movie = ' '.join(command.split()[2:])
```
Comment (jonheese-wti) [inline plugins/tmdb.py:23]: You already `split()` the incoming text string in L21, so you don't need to (and can't) do it again here.

File: plugins/tmdb.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            movie_base = 'https://www.themoviedb.org/movie'
+            media_base = api_config['images']['secure_base_url']
+
+            # Get the smallest image, slack will still crop it
+            # maybe add in flight resizing at some point -CC
+            image_size = api_config['images']['poster_sizes'][0]
+
+            result_num = 0
+
+            if lookup['total_result'] == 0:
```
Comment (jonheese-wti) [inline plugins/tmdb.py:57]: Looks like this should be `total_results` (plural), I assume this is a typo.

**Comment by jonheese-wti** [comment]
After the fixes mentioned above, this will be ready to merge. 👍

File: plugins/tmdb.py (hunk 4 above)
Comment (cacoyle) [reply plugins/tmdb.py]: D'oh, you are correct sir

File: plugins/tmdb.py (hunk 5 above)
Comment (cacoyle) [reply plugins/tmdb.py]: Also correct

**Pull request merged**